monitoring/src/main/java/org/polypheny/db/monitoring/events/analyzer/DMLEventAnalyzer.java

```
@@ -0,0 +1,98 @@
+    // TODO: Bis jetzt sind die Klassen mehr oder weniger identisch. Ist das einfach vorbereitet für später oder wie?
+
+
+    public static DMLDataPoint analyze( DMLEvent dmlEvent ) {
+        DMLDataPoint metric = DMLDataPoint
+                .builder()
+                .description( dmlEvent.getDescription() )
+                .Id( dmlEvent.getId() )
+                .fieldNames( dmlEvent.getFieldNames() )
+                .executionTime( dmlEvent.getExecutionTime() )
+                .rowCount( dmlEvent.getRowCount() )
+                .isSubQuery( dmlEvent.isSubQuery() )
+                .recordedTimestamp( dmlEvent.getRecordedTimestamp() )
+                .build();
+
+        RelRoot relRoot = dmlEvent.getRouted();
+        if ( relRoot != null ) {
+            RelNode node = relRoot.rel;
+            processRelNode( node, dmlEvent, metric );
+        }
+
+        if ( dmlEvent.isAnalyze() ) {
+            processDurationInfo( dmlEvent, metric );
+        }
+
+        return metric;
+    }
+
+
+    private static void processDurationInfo( DMLEvent dmlEvent, DMLDataPoint metric ) {
+        try {
+            InformationDuration duration = new Gson().fromJson( dmlEvent.getDurations(), InformationDuration.class );
+            getDurationInfo( metric, "Plan Caching", duration );
+            getDurationInfo( metric, "Index Lookup Rewrite", duration );
+            getDurationInfo( metric, "Constraint Enforcement", duration );
+            getDurationInfo( metric, "Implementation Caching", duration );
+            getDurationInfo( metric, "Index Update", duration );
+            getDurationInfo( metric, "Routing", duration );
+            getDurationInfo( metric, "Planning & Optimization", duration );
+            getDurationInfo( metric, "Implementation", duration );
+            getDurationInfo( metric, "Locking", duration );
+        } catch ( Exception e ) {
+            log.debug( "could not deserialize of get duration info" );
```

This more or less ignores the exception...